src/pools/selection-strategies.ts

```diff
@@ -136,15 +136,12 @@ class DynamicPoolWorkerChoiceStrategy<Worker extends IWorker, Data, Response>
 
   /** @inheritdoc */
   public choose (): Worker {
-    const freeWorker = SelectionStrategiesUtils.findFreeWorkerBasedOnTasks(
-      this.pool.tasks
-    )
+    const freeWorker = this.pool.findFreeTasksMapEntry()[0]
```

Comment:
Mh :thinking:  
Maybe `null` instead of `false` would be better, I'm not sure

Reply:
You can't return a restructured array and a boolean, the code need to access each member of the array.

Reply:
Please check my PR #243